bbq/compiler/compiler.go

```diff
@@ -1589,9 +1591,42 @@ func (c *Compiler[_, _]) VisitBinaryExpression(expression *ast.BinaryExpression)
 	return
 }
 
-func (c *Compiler[_, _]) VisitFunctionExpression(_ *ast.FunctionExpression) (_ struct{}) {
-	// TODO
-	panic(errors.NewUnreachableError())
+func (c *Compiler[_, _]) VisitFunctionExpression(expression *ast.FunctionExpression) (_ struct{}) {
+	// TODO: desugar function expressions
```

Comment:
I guess until they are desugar-ed, the pre/post conditions wouldn't be supported?

Reply:
Correct. I think I tried it and an assertion triggered that the function still had conditions